### PR TITLE
Scc 3178 integrate date parsing in batches

### PR DESCRIPTION
**Pull request by charmingduchess**
Incorporate date parsing into the discovery hybrid indexer
- update tests with AWS stubbing
- volume parsing integration to come!

### CODE REVIEW

File: lib/date-parse/index.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+const parseDatesAndCache = async function (bibs) {
+  const dates = extractFieldtagvs(bibs)
+  const preparsedDates = preparse(dates)
+  try {
+    let ranges = await timeTwist(preparsedDates)
+    ranges = filterNulls(ranges, preparsedDates)
+    dateCache = dates.reduce((cache, date, i) => {
+      return { ...cache, [date]: ranges[i] }
+    }, {})
```
Comment (nonword) [inline lib/date-parse/index.js:85]: Can we add time-ellapsed reporting to this call? I predict we'll want to be able to inspect slow calls. Something like:
```
logger.debug(`parseDatesAndCache: Took {time}ms to process ${dates.length} from ${bibs.length} bibs`)
```

Comment (nonword) [reply lib/date-parse/index.js:85]: Lines 79 - 82 are basically exactly what `_parseDates` does. Couldn't you define `parseDatesAndCache` like this?
```
const parseDatesAndCache = async function (bibs) {
  const dates = extractFieldtagvs(bibs)
  const ranges = _parseDates(dates)
  
  // Update cache:
  dateCache = dates.reduce((cache, date, i) => {
    return { ...cache, [date]: ranges[i] }
  }, {})
}
```

Comment (charmingduchess) [reply lib/date-parse/index.js:83]: updated!

**Review by nonword — COMMENTED** [review]
Looks good. But I think we realized this morning that the date parsing logic needs to be returned to the discovery-store-poster?

File: package.json [hunk 2]
```diff
@@ -18,8 +18,7 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "./node_modules/.bin/standard && ./node_modules/.bin/mocha test --grep '@local-only' --invert",
-    "integration-test": "./node_modules/.bin/mocha test --grep '@local-only'"
+    "test": "./node_modules/.bin/standard && ./node_modules/.bin/mocha test --grep '@local-only' --invert"
```
Comment (nonword) [inline package.json:21]: let's not drop `standard`

**Review by nonword — APPROVED** [review]

**Pull request merged**